pkg/group/group_integration_test.go

```diff
@@ -0,0 +1,196 @@
+package group_test
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/dhis2-sre/im-manager/pkg/group"
+	"github.com/dhis2-sre/im-manager/pkg/inttest"
+	"github.com/dhis2-sre/im-manager/pkg/model"
+	"github.com/dhis2-sre/im-manager/pkg/user"
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/require"
+)
+
+type TestAuthenticationMiddleware struct{}
+
+func (t TestAuthenticationMiddleware) TokenAuthentication(c *gin.Context) {}
+
+type TestAuthorizationMiddleware struct{}
+
+func (t TestAuthorizationMiddleware) RequireAdministrator(c *gin.Context) {
+	c.Next()
+}
+
+func TestGroupHandler(t *testing.T) {
+	t.Parallel()
+
+	db := inttest.SetupDB(t)
+	userRepository := user.NewRepository(db)
+	userService := user.NewService(userRepository)
+	groupRepository := group.NewRepository(db)
+	groupService := group.NewService(groupRepository, userService)
+
+	err := user.CreateAdminUser("admin", "admin", userService, groupService)
+	require.NoError(t, err, "failed to create admin user and group")
+
+	client := inttest.SetupHTTPServer(t, func(engine *gin.Engine) {
+		handler := group.NewHandler(groupService)
+		authentication := TestAuthenticationMiddleware{}
+		authorization := TestAuthorizationMiddleware{}
+		group.Routes(engine, authentication, authorization, handler)
+	})
+
+	var userId string
+	var user *model.User
+	{
+		user, err = userService.FindOrCreate("user@dhis2.org", "oneoneoneoneoneoneone111")
+		require.NoError(t, err)
+		userId = strconv.FormatUint(uint64(user.ID), 10)
+	}
+
+	var groupName string
+	{
+		group, err := groupService.Create("test-group", "test-hostname.com", true)
+		require.NoError(t, err)
+		groupName = group.Name
+	}
+
+	t.Run("CreateGroup", func(t *testing.T) {
+		t.Parallel()
+
+		t.Run("Deployable", func(t *testing.T) {
+			requestBody := strings.NewReader(`{
+				"name": "deployable-test-group",
+				"hostname": "deployable-test-hostname.com",
+				"deployable": true
+			}`)
+
+			var group model.Group
+			client.PostJSON(t, "/groups", requestBody, &group)
+			require.Equal(t, "deployable-test-group", group.Name)
+			require.Equal(t, "deployable-test-hostname.com", group.Hostname)
+			require.Equal(t, true, group.Deployable)
+		})
+
+		t.Run("NonDeployable", func(t *testing.T) {
+			requestBody := strings.NewReader(`{
+				"name": "non-deployable-test-group",
+				"hostname": "non-deployable-test-hostname.com",
+				"deployable": false
+			}`)
+
+			var group model.Group
+			client.PostJSON(t, "/groups", requestBody, &group)
+			require.Equal(t, "non-deployable-test-group", group.Name)
+			require.Equal(t, "non-deployable-test-hostname.com", group.Hostname)
+			require.Equal(t, false, group.Deployable)
+		})
+	})
+
+	t.Run("AddUserToGroup", func(t *testing.T) {
+		t.Parallel()
+
+		t.Run("AddUserToGroup", func(t *testing.T) {
```

Comment:
You are repeating the parents subtest name. We could call it

```suggestion
		t.Run("Success", func(t *testing.T) {
```

like we do in the user intttest. This can also serve as an easy to remember filter for running the test alone.

Same for the other subtests. The names are concatenated in `subtest1/subtest2/...`. I would therefore not repeat the parents name but try to make it read more like a sentence.